drivers/volume/portworx/portworx.go

```diff
@@ -4314,17 +4342,22 @@ func (p *portworx) createStatfsConfigMap(cmNamespace string) error {
 			statfsSOName: statfsSOContents,
 		},
 		Data: map[string]string{
-			"ld.so.preload": path.Join(statfsSODirInVirtLauncher, statfsSOName),
+			ldPreloadFileName: statfsSOPathInVirtLauncher,
 		},
 	}
+}
 
-	if _, err := core.Instance().CreateConfigMap(cm); err != nil && !k8s_errors.IsAlreadyExists(err) {
-		logrus.Errorf("Failed to create  config map %v/%v: %v", cmNamespace, statfsConfigMapName, err)
-		return err
-	} else if err == nil {
-		logrus.Infof("Created configmap %v/%v", cmNamespace, statfsConfigMapName)
+func (p *portworx) statfsConfigMapNeedsUpdate(existing *v1.ConfigMap, statfsSOContents []byte) bool {
+	existingBytes := existing.BinaryData[statfsSOName]
+	if len(existingBytes) != len(statfsSOContents) {
+		return true
 	}
-	return nil
+	for i := range existingBytes {
+		if existingBytes[i] != statfsSOContents[i] {
```

Comment:
Should we calculate an md5sum and compare that instead of the byte content? The downside is we will end up updating the config map the first time for everyone who upgrades from a version which does not have an md5sum.

Reply:
I think it's a good idea. 
We can also use hashstructure library as used here https://github.com/libopenstorage/stork/blob/b6e03704b305d2c601dd2fb7d420398ba9b98103/pkg/migration/controllers/migration.go#L1305C2-L1305C2 
Downside will only come in picture if we store the hash of existing CM and not when do both conversions on the fly. Even then it should be ok.

Reply:
Added sha256. In hindsight, this is a bit of an overkill since the file size is only 20k. But might help in future if the file size grows.